packages/core/src/plugins/mathDelimiters.js

```diff
@@ -0,0 +1,16 @@
+const katex = require('katex');
+const texmath = require('markdown-it-texmath');
+const md = require('../lib/markdown-it');
+
+// see https://github.com/goessner/markdown-it-texmath#features for supported delimiters
+// note that some of them may not work correctly due to conflicting syntax
+function mathDelimiters(pluginContext) {
+  if (!pluginContext.delimiters) {
+    throw new Error('No delimiters defined in the plugin context!');
+  }
+  md.use(texmath, { engine: katex, delimiters: pluginContext.delimiters });
```

Comment:
Neat method of adding the md plugin, didn't think of this in considering https://github.com/MarkBind/markbind/issues/1663#issuecomment-1082087909 (ignore that comment). 🤦‍♂️ 

Would this work ok with `markbind serve`? (`beforeSiteGenerate` is called multiple times)

If not maybe we could gate this behind a 

```
let pluginAdded = false
if (!pluginAdded) {
  ...
  pluginAdded = true
}
```

flag of sorts

Reply:
I think loading it a few times doesn't seem to break anything, but I guess it might be a problem depending on how markdown-it handles plugin loading (I'm not sure). I have added the flag as suggested.